e621/endpoints/my_endpoints.py

```diff
@@ -1,18 +1,213 @@
-from e621 import models
+from typing import List, Optional
+
+from e621.models import EnrichedPool, Note, Tag, TagAlias
+
 from .endpoints import BaseEndpoint
 
 
 class Tags(BaseEndpoint):
-    pass
+    def get(self, tag_id: int) -> Tag:
+        return Tag(**self._api.session.request("GET", f"tags/{tag_id}.json").json())
+
+    def search(
+        self,
+        name_matches: Optional[str] = None,
+        category: Optional[str] = None,
+        order: Optional[str] = None,
+        hide_empty: Optional[bool] = None,
+        has_wiki: Optional[bool] = None,
+        has_artist: Optional[bool] = None,
+        limit: Optional[int] = None,
+        page: Optional[str] = None,

```

Comment:
> Technically str is the correct data type because we can send either a formatted a/b + tag_id or a page number. However, we need to check and think about a few things:
> 
> 1. How well e621 api handles strings in place of ints if we're sending the page number
> 2. Do we want to specify and validate what data the user sends. For example, we could make the typehint for pages `Union[str, int]` or even make it `Union[TagOffset, PageNumber, None]` where PageNumber is a TypeAlias for int and TagOffset is a separate class that validates the correct format for a/b + tag_id. So it just depends on how much do we want to hold the user's hand.

I think we should validate what we can validate, so I prefer `Union[PageOffset, PageNumber, None]`.
Moreover, `Union[PageOffset, PageNumber, None]` documents itself better than `Union[str, int, None]` which literally tells no information about its variants except types.

Write `PageOffset(OffsetRelation.BEFORE, 1)` instead of `b1` is not a great deal, but leaves no place for such hard to trace typos as `d1`, etc.

Reply:
Agreed. Let's do this after we merge both pull requests though.